src/Components/Filter/Filter.js

```diff
@@ -0,0 +1,208 @@
+import React from "react";
+import { categories } from "../../backend/db/categories";
+import { products } from "../../backend/db/products";
+import { useProduct } from "../../contexts/ProductProvider";
+
+export const getSortedData = (state, productList) => {
```

Comment:
you can create utility functions out of these and move them to separate files to make them more generic. do it if you have time

Reply:
Will do it at the end.